src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java

```diff
@@ -112,6 +112,71 @@ public void insertDocumentWithNumberId() {
           .body("data.docs[0]", jsonEquals(expected));
     }
 
+    @Test
+    public void insertDuplicateDocument() {
+      String json =
+          """
+                            {
+                              "insertOne": {
+                                "document": {
+                                  "_id": "duplicate",
+                                  "username": "user4"
+                                }
+                              }
+                            }
+                            """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.insertedIds[0]", is("duplicate"));
+
+      json =
+          """
+                  {
+                    "find": {
+                      "filter" : {"_id" : "duplicate"}
+                    }
+                  }
+                  """;
+      String expected = "{\"_id\": \"duplicate\", \"username\":\"user4\"}";
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.docs[0]", jsonEquals(expected));
+
+      json =
+          """
+                {
+                  "insertOne": {
+                    "document": {
+                      "_id": "duplicate",
+                      "username": "user4"
```

Comment:
Maybe try inserting document with different `username`, and after failure make sure nothing was changed?
(i.e. no Upsert happened)